### PR TITLE
Add Release Channel selection in-product

### PR DESCRIPTION
**Pull request by iethree**
part of https://github.com/metabase/metabase/issues/48121

### Description

Adds a new setting: `update-channel` in admin settings and shows appropriate release notes for it

Screen Shot 2024-09-25 at 1 55 04 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: src/metabase/public_settings.clj [hunk 1]
```diff
@@ -71,6 +71,24 @@
...
+    (setting/set-value-of-type! :string :update-channel new-channel)))
+
+(defsetting update-channel
+  (deferred-tru "Select an update channel to customize which Metabase releases you receive notifications for.")
+  :visibility :admin
+  :type       :string
+  :encryption :no
+  :audit      :getter
+  :setter     set-update-channel!
+  :default    "latest")
```
Comment (iethree) [inline src/metabase/public_settings.clj:90]: look I'm a clojure dev!

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  type: "select",
+  description:
+    "Select an update channel to customize which Metabase releases you receive notifications for.",
+  defaultValue: "latest",
+  options: [
+    { name: "Stable", value: "latest" },
+    { name: "Beta", value: "beta" },
+    { name: "Nightly", value: "nightly" },
+  ],
+};
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx:25]: @jeff-bruemmer , would appreciate your eyes on this copy. this also probably needs some documentation.

see our [internal documentation](https://www.notion.so/metabase/Release-Channel-Tagging-437bdfa7b0c54579badd058a84a64c85) on this:

File: src/metabase/public_settings.clj [hunk 3]
```diff
@@ -71,6 +71,25 @@
...
+
+(defsetting update-channel
+  (deferred-tru "Select an update channel to customize which Metabase releases you receive notifications for.")
+  :visibility :admin
+  :type       :string
+  :encryption :no
+  :export?    true
+  :audit      :getter
+  :setter     set-update-channel!
+  :default    "latest")
```
Comment (iethree) [inline src/metabase/public_settings.clj:91]: @metabase/core-backend-admin-webapp please look at this extra closely - this is not my native tongue

File: src/metabase/public_settings.clj [hunk 4]
```diff
@@ -71,6 +71,25 @@
   :audit   :getter
   :default true)
 
+(defn- set-update-channel! [new-channel]
+  (let [valid-channels ["latest" "beta" "nightly"]]
+    ;; check to make sure `update-channel` is a valid channel, or throw an Exception it is it not.
+    (when-not (some #{new-channel} valid-channels)
```
Comment (tsmacdonald) [inline src/metabase/public_settings.clj:77]:
```suggestion
  (let [valid-channels #{"latest" "beta" "nightly"}]
    (when-not (valid-channels new-channel)
```

not a high-conviction change, but saves a couple characters and probably a couple clock cycles

Comment (tsmacdonald) [reply src/metabase/public_settings.clj:77]: (and I don't think the comment is helpful)

File: src/metabase/public_settings.clj (hunk 3 above)
Comment (tsmacdonald) [reply src/metabase/public_settings.clj:91]: Pop this into the bottom of `test/metabase/public_settings_test.clj`

```
(deftest update-channel-test
  (testing "we can set the update channel"
    (mt/discard-setting-changes [update-channel]
      (public-settings/update-channel! "nightly")
      (is (= "nightly" (public-settings/update-channel)))))
  (testing "we can't set the update channel to an invalid value"
    (mt/discard-setting-changes [update-channel]
      (is (thrown?
           IllegalArgumentException
           (public-settings/update-channel! "millennially"))))))
```

**Review by tsmacdonald — APPROVED** [review]
BE looks okay, please consider the changes I suggested though

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import { useSetting } from "metabase/common/hooks";
+import CS from "metabase/css/core/index.css";
+import { Flex } from "metabase/ui";
+import type { SettingDefinition, UpdateChannel } from "metabase-types/api";
+
+import { SettingsSetting } from "../SettingsSetting";
+
+import { VersionUpdateNotice } from "./VersionUpdateNotice/VersionUpdateNotice";
+
+const updateChannelSetting = {
```
Comment (npfitz) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx:14]: Not worth blocking a merge, but is there a reason this wasn't defined in `selectors.js`?

**Review by npfitz — COMMENTED** [review]
Can you confirm that this is working as expected? I didn't see any tests around this. I know that some of these values from from `version.properties`, which may be changing?

chrome_qD68DudMRI

**Review by npfitz — APPROVED** [review]
Chatted on Slack. Looks good 👍

**Review by npfitz — APPROVED** [review]
:chef-kiss:

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  key: "update-channel",
+  display_name: "Update Channel",
+  type: "select",
+  description:
+    "Select an update channel to customize which Metabase releases you receive notifications for.",
```
Comment (jeff-bruemmer) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx:19]: How about:

```suggestion
    "Metabase will send you updates when a new release is available for the channel you select.",
```

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  defaultValue: "latest",
+  options: [
+    { name: "Stable", value: "latest" },
```
Comment (jeff-bruemmer) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx]: Just so I understand how this is expected to work:

Technically, stable could refer to older (stable) releases, right? Will people get notified of the latest stable release on their current version (e.g., when we backport changes to a previous release), or only the latest and greatest release?

Given the value is "latest" here: I'm assuming we only notify on the latest releases (as currently defined), so we might want to clarify that in the UI, just to make sure people don't think that when they see latest in another context that we're referring to an upstream build like nightly or beta.

```suggestion
    { name: "Stable (latest)", value: "latest" },
```
If people can be on a downstream stable release, and will get notified of point releases on that version, the original copy is preferred, as stable won't always mean latest.

Comment (iethree) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx]: > stable won't always mean latest

it should. we should never have a beta or nightly build tagged as latest.

Comment (iethree) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm/SettingsUpdatesForm.tsx]: >  Will people get notified of the latest stable release on their current version (e.g., when we backport changes to a previous release), or only the latest and greatest release?

There is only ever one latest/stable release. If you are on 49.1 today, you will get notified that you should upgrade to 50.28 (even though there exists a 49.25) Similarly, if you are on 50.27, you will be notified to upgrade to 50.28.

**Pull request merged**